osu.Game/Rulesets/Objects/Pooling/PoolableDrawableWithLifetime.cs

```diff
@@ -100,11 +100,7 @@ private void setLifetime(double start, double end)
             base.LifetimeStart = start;
             base.LifetimeEnd = end;
 
-            if (Entry != null)
-            {
-                Entry.LifetimeStart = start;
-                Entry.LifetimeEnd = end;
-            }
+            Entry?.SetLifetime(start, end);
```

Comment:
Am I right in thinking that this is a critical importance? I imagine that setting LifetimeStart or LifetimeEnd would break it, with the case:

```
KeepAlive = true
realLifetimeStart = SomeValueA
realLifetimeEnd = SomeValueB

LifetimeStart = start
realLifetimeStart = start
realLifetimeEnd = double.MaxValue (note loss of SomeValueB)
```
If I'm correct, I think this should be documented as such.

Reply:
I didn't think of that case, but I think you are right.
Not sure how to solve this issue, though.
The logic is weird as `entry.LifetimeStart = entry.LifetimeStart` is not a no-op.
I'm not even sure where to put the comment because `PoolableDrawableWithLifetime` shouldn't care about `HitObjectLifetimeEntry`-specific behavior.

Reply:
How about splitting up `SetLifetime()` into two `SetLifetimeStart()` and `SetLifetimeEnd()`?